Parallel build is broken for STATIC_BUILD=ON

### How to reproduce

```console
$ export MAKEFLAGS=j
$ tarantoolctl rocks install kafka STATIC_BUILD=ON
...
make[2]: *** No rule to make target 'librdkafka/lib/librdkafka.a', needed by 'kafka/tntkafka.so'. Stop.
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

### How it affects others

In the other project (central) `rocks install` is called from the top-level cmake build. There we install multiple rocks so we want to do it in parallel (like `cmake && make -j`). Cmake sets the MAKEFLAGS environment variable thus building kafka fails.

--- a/.github/workflows/fast_testing.yml
+++ b/.github/workflows/fast_testing.yml
@@ -48,7 +48,9 @@ jobs:
         run: pip3 install -r tests/requirements.txt
 
       - name: Build module
-        run: tarantoolctl rocks STATIC_BUILD=ON make
+        run: |
+          export MAKEFLAGS=-j8
+          tarantoolctl rocks STATIC_BUILD=ON make
 
       - name: Run tarantool application
         run: TT_LOG=tarantool.log tarantool tests/app.lua > output.log 2>&1 &

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,8 +10,7 @@ set(MY_C_FLAGS "-Wall -Wextra -Werror -std=gnu11 -fno-strict-aliasing -Wno-depre
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MY_C_FLAGS}")
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} ${MY_C_FLAGS} -ggdb3")
 
-set(Tarantool_FIND_REQUIRED ON)
-find_package(Tarantool)
+find_package(Tarantool REQUIRED)
 # Oracle Connector requires fiber_cond API added in 1.7.4-291
 string(REPLACE "-" "." tntver ${TARANTOOL_VERSION})
 if(${tntver} VERSION_LESS 1.7.4.291)
@@ -42,9 +41,12 @@ if(STATIC_BUILD)
         DEPENDS ${CMAKE_BINARY_DIR}/librdkafka/lib/librdkafka.a
     )
 
+    add_library(librdkafka_static INTERFACE)
+    target_include_directories(librdkafka_static INTERFACE ${CMAKE_BINARY_DIR}/librdkafka/include)
+    target_link_libraries(librdkafka_static INTERFACE ${CMAKE_BINARY_DIR}/librdkafka/lib/librdkafka.a)
+    add_dependencies(librdkafka_static librdkafka.a)
 
-    set(RDKAFKA_LIBRARY  ${RDKAFKA_LIBRARY} ${CMAKE_BINARY_DIR}/librdkafka/lib/librdkafka.a)
-    set(RDKAFKA_INCLUDE_DIR ${CMAKE_BINARY_DIR}/librdkafka/include)
+    set(RDKAFKA_LIBRARY ${RDKAFKA_LIBRARY} librdkafka_static)
 else()
     set(RDKAFKA_FIND_REQUIRED ON)
     find_package(RdKafka)

